scripts/completion/bash_autocomplete

```diff
@@ -0,0 +1,32 @@
+#! /bin/bash
+
+# Macs have bash3 for which the bash-completion package doesn't include
+# _init_completion. This is a minimal version of that function.
+_cli_init_completion() {
+  COMPREPLY=()
+  _get_comp_words_by_ref "$@" cur prev words cword
+}
+
+_cli_bash_autocomplete() {
+  if [[ "${COMP_WORDS[0]}" != "source" ]]; then
+    local cur opts base words
+    COMPREPLY=()
+    cur="${COMP_WORDS[COMP_CWORD]}"
+    if declare -F _init_completion >/dev/null 2>&1; then
+      _init_completion -n "=:" || return
+    else
+      _cli_init_completion -n "=:" || return
+    fi
+    words=("${words[@]:0:$cword}")
+    if [[ "$cur" == "-"* ]]; then
+      requestComp="${words[*]} ${cur} --generate-bash-completion"
+    else
+      requestComp="${words[*]} --generate-bash-completion"
+    fi
+    opts=$(eval "${requestComp}" 2>/dev/null)
+    COMPREPLY=($(compgen -W "${opts}" -- ${cur}))
+    return 0
```

Comment:
```suggestion
    # If a single completion is found, append a space automatically
    if [[ ${#COMPREPLY[@]} -eq 1 ]]; then
      COMPREPLY[0]+=" "
    fi
    return 0
```

Reply:
![image](https://github.com/user-attachments/assets/32f7bb07-4199-4c02-8e66-0ca7e61385d2)

Reply:
We can do separate script for curio and sptool if required. Use a hardcoded --t01000 for sptool to generate the completions.

Reply:
Setting the environment variable `LOTUS_DOCS_GENERATION=1` can resolve this issue.
https://github.com/filecoin-project/curio/blob/d9a219debf0facfb506b3a4d9bd3de36a9fcacb7/cmd/sptool/main.go#L46-L49

![image](https://github.com/user-attachments/assets/bccdc94f-5c49-4579-aba9-a993d4f8ac6f)

Reply:
Setting that var in prod is not possible. It would affect other parts of the build. Instead, can we try to set it in the script itself and see if we get the output we need?

Reply:
Maybe I expressed it incorrectly. This variable already exists; I just added it in the script. The curio code hasn't changed.

Reply:
I see. That was confusing. I did not look at the commit.

Reply:
Sorry about the force-pushed.
https://github.com/filecoin-project/curio/compare/2ed3755e9fe10ed65804dcf8815835ababd3cd76..050e4c0e22f5eb5c9047eb1e113bcbb12d24b34d

Reply:
All good. Thank you for the contribution.
Approved and merged.